source/math/FloatVectorList.ooc

```diff
@@ -306,6 +306,12 @@ FloatVectorList: class extends VectorList<Float> {
 			result = this copy()
 		result
 	}
+	scalarProduct: func (other: This) -> Float {
+		result := 0.f
+		for (i in 0 .. this count minimum(other count))
+			result += this[i] * other[i]
+		result
```

Comment:
Yes, but then you have to instantiate a new `VectorList` that you have to free. This seems like a useful function to have to me.